crates/syn-solidity/src/yul/ident/mod.rs

```diff
@@ -0,0 +1,126 @@
+use crate::Spanned;
+
+use proc_macro2::{Ident, Span};
+use quote::ToTokens;
+use std::fmt;
+use syn::{
+    ext::IdentExt,
+    parse::{Parse, ParseStream},
+    Result,
+};
+
+mod path;
+pub use path::YulPath;
+
+/// A Yul identifier.
+///
+/// Solidity Reference:
+/// <https://docs.soliditylang.org/en/latest/grammar.html#a4.SolidityLexer.YulIdentifier>
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct YulIdent(pub Ident);
```

Comment:
Are these different in any way to the regular Solidity ones? The grammar def is the same, but I'm wondering if the keyword handling is different that it merits a separate type

Reply:
Solidity and Yul handle keywords differently because they each have their own unique sets of keywords and reserved identifiers. 

[solidity reserved  identifiers:](https://github.com/ethereum/solidity/blob/020b59680ebce25367ece2c1388584a8c3827d85/liblangutil/Token.h#L235) `after`, `alias`, `apply`, `auto`, `byte`, ...

[yul reserved identifiers:](https://github.com/ethereum/solidity/blob/020b59680ebce25367ece2c1388584a8c3827d85/libyul/backends/evm/EVMDialect.cpp#L114) `basefee`, `prevrandao`, `linkersymbol`, `datasize`, `dataoffset`, `datacopy`, ...

[solidity keywords](https://github.com/ethereum/solidity/blob/020b59680ebce25367ece2c1388584a8c3827d85/liblangutil/Token.h#L142)

[yul keywords](https://github.com/ethereum/solidity/blob/020b59680ebce25367ece2c1388584a8c3827d85/liblangutil/Token.h#L321) 

Are keyword checks planned for implementation in the future? if not, then can reuse the Solidity ident type for Yul.

Reply:
Thanks for the links, I'm still thinking about it. This wrapper type is fine for now

Reply:
Leaving it as is for now. The code duplication feels a little bit off, but curious to hear what you come up with